packages/segmented-control/src/SegmentedControlOption.tsx

```diff
@@ -0,0 +1,238 @@
+  cursor: pointer;
+  transition: all 100ms ease-in-out;
+  z-index: 1;
+
+  &:hover {
+    color: var(--hover-text-color);
+
+    &:after {
+      background-color: var(--hover-background-color);
+    }
+
+    &${checkedSelector}, &[data-disabled='true'] {
+      border-color: transparent;
+      background-color: transparent;
+
+      &:after {
+        background-color: transparent;
+      }
+    }
+  }
+
+  &${checkedSelector} {
+    color: var(--active-text-color); // color
+  }
+
+  &[data-disabled='true'] {
+    color: var(--disabled-text-color); // color
+    cursor: not-allowed;
+  }
+
+  // Hover indicator
+  &:after {
+    content: '';
+    position: absolute;
+    height: calc(100% - 2px);
+    width: 100%;
+    top: 1px;
+    left: 0;
+    background-color: transparent;
+    z-index: -1;
+    border-radius: inherit;
+    transition: all 100ms ease-in-out;
+  }
+
+  /* 
+   * Adds the divider line to unselected segments 
+   */
+  &:before {
+    content: '';
+    position: absolute;
+    height: var(--divider-height);
+    width: 1px;
+    left: calc(0px - (var(--segment-gap, 1px) + 1px) / 2);
+    background-color: transparent;
+    transition: all 100ms ease-in-out;
+  }
+
+  &${uncheckedSelector} {
+    &:not(:first-child) {
+      &:not(${checkedSelector} + ${uncheckedSelector}) {
+        // no divider to the left of the checked segment
+        &:before {
+          background-color: ${uiColors.gray.light1}; // color
+        }
+      }
+    }
+  }
+`;
+
+const radioInputStyle = css`
+  height: 0;
+  width: 0;
+  margin: 0;
+  opacity: 0;
+  pointer-events: none;
+`;
+
+const labelStyle = css`
+  display: inline-flex;
+  align-items: center;
+  gap: calc(var(--font-size) / 2);
+  cursor: inherit;
+  z-index: 3;
+`;
+
+/**
+ * Types
+ */
+
+// Internal “Private” Props
+// _name - the name of the segmented control. Used to group the HTML radio inputs
+// _checked - whether the option is checked. Defined by the parent
+// _onChange - the onChange handler passed in from the SegmentedControl
+export interface SegmentedControlOptionProps {
+  value: string;
+  children: React.ReactNode;
+  disabled?: boolean;
+  as?: any;
+  className?: string;
+  id?: string;
+  size?: Size;
+  darkMode?: boolean;
+  _name?: string;
+  _checked?: boolean;
+  _onChange?: React.ChangeEventHandler<HTMLInputElement>;
+}
+
+/**
+ * Component
+ */
+const SegmentedControlOption = React.forwardRef<
+  HTMLLabelElement,
+  SegmentedControlOptionProps
+>(function SegmentedControlOption(
+  {
+    value,
+    children,
+    disabled = false,
+    as = 'div', // TODO
+    className,
+    id,
+    size = 'default',
+    darkMode = false,
+    _onChange: onChange,
+    _name: name,
+    _checked: checked,
+  }: SegmentedControlOptionProps,
+  forwardedRef,
+) {
+  const mode = darkMode ? 'dark' : 'light';
+
+  return (
+    <label
+      htmlFor={id}
+      className={cx(optionStyle({ mode, size }), className)}
+      data-disabled={`${disabled}`}
```

Comment:
I saw this in a selector above as well. We probably shouldn't be using a data prop for this. Let me know if you'd like help brainstorming alternatives.